crypto/pedersen/pedersen.go

```diff
@@ -0,0 +1,56 @@
+package pedersen
+
+import (
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"os"
+	"os/exec"
+)
+
+// Requires to follow instructions to install pycrypto: https://github.com/stefandeml/zokrates-pycrypto
+// And either set CENT_PED_BIN with cli.py location or store the repo in $HOME/zksnarks/pycrypto
+
+// NewPedersenHash returns a pedersen hash function
+func NewPedersenHash() hash.Hash {
+	// Resolve python binary location
+	pedBin := os.Getenv("CENT_PED_BIN")
+	if pedBin == "" {
+		pedBin = fmt.Sprintf("%s/zksnarks/pycrypto/cli.py", os.Getenv("HOME"))
+	}
+	return &pedersen{binLocation: pedBin}
+}
+
+type pedersen struct {
+	binLocation string
+	hashed      []byte
+}
+
+func (pd *pedersen) Reset() {
+	pd.hashed = []byte{}
+}
+
+func (pd *pedersen) Write(p []byte) (n int, err error) {
+	args := []string{pd.binLocation, "hash", hex.EncodeToString(p)}
+	o, err := exec.Command("python3", args...).Output()
```

Comment:
Temporary until we have a go implementation of this hashing function.

Reply:
we have to update the docker to have the python 3 executable then